monitor.py

```diff
@@ -411,197 +467,258 @@ def parse_entry(self, line):
                     self.systempopulation = entry.get('Population')
 
-                (self.system, self.station) = (entry['StarSystem'] == 'ProvingGround' and 'CQC' or entry['StarSystem'],
-                                               entry.get('StationName'))	# May be None
-                self.station_marketid = entry.get('MarketID') # May be None
-                self.stationtype = entry.get('StationType')	# May be None
-                self.stationservices = entry.get('StationServices')	# None under E:D < 2.4
-            elif entry['event'] == 'ApproachBody':
+                self.system = 'CQC' if entry['StarSystem'] == 'ProvingGround' else entry['StarSystem']
+                self.station = entry.get('StationName')  # May be None
+                self.station_marketid = entry.get('MarketID')  # May be None
+                self.stationtype = entry.get('StationType')  # May be None
+                self.stationservices = entry.get('StationServices')  # None under E:D < 2.4
+
+            elif event_type == 'ApproachBody':
                 self.planet = entry['Body']
-            elif entry['event'] in ['LeaveBody', 'SupercruiseEntry']:
+
+            elif event_type in ('LeaveBody', 'SupercruiseEntry'):
                 self.planet = None
 
-            elif entry['event'] in ['Rank', 'Promotion']:
+            elif event_type in ('Rank', 'Promotion'):
                 payload = dict(entry)
                 payload.pop('event')
                 payload.pop('timestamp')
-                for k,v in payload.items():
-                    self.state['Rank'][k] = (v,0)
-            elif entry['event'] == 'Progress':
-                for k,v in entry.items():
-                    if k in self.state['Rank']:
-                        self.state['Rank'][k] = (self.state['Rank'][k][0], min(v, 100))	# perhaps not taken promotion mission yet
-            elif entry['event'] in ['Reputation', 'Statistics']:
+
+                self.state['Rank'].update({k: (v, 0) for k, v in payload.items()})
+
+            elif event_type == 'Progress':
+                rank = self.state['Rank']
+                for k, v in entry.items():
+                    if k in rank:
+                        # perhaps not taken promotion mission yet
+                        rank[k] = (rank[k][0], min(v, 100))
+
+            elif event_type in ('Reputation', 'Statistics'):
                 payload = OrderedDict(entry)
                 payload.pop('event')
                 payload.pop('timestamp')
-                self.state[entry['event']] = payload
+                self.state[event_type] = payload
+
+            elif event_type == 'EngineerProgress':
+                engineers = self.state['Engineers']
+                if 'Engineers' in entry:  # Startup summary
+                    self.state['Engineers'] = {
+                        e['Engineer']: (e['Rank'], e.get('RankProgress', 0))
+                        if 'Rank' in e else e['Progress'] for e in entry['Engineers']
```

Comment:
It feels like the conditional should be indented to emphasise it's attached to the previous line.

Reply:
Oh whoops, I assumed that was part of the comprehension